File: app/src/main/java/com/handstandsam/maintainableespresso/mockaccount/MockAccount.java
package com.handstandsam.maintainableespresso.mockaccount;


import com.handstandsam.maintainableespresso.models.Category;
import com.handstandsam.maintainableespresso.models.Item;
import com.handstandsam.maintainableespresso.models.User;

import java.util.ArrayList;
import java.util.List;

public abstract class MockAccount {

    public abstract String getUsername();

    public abstract User getUser();

    public List<Category> getCategories(){
        return new ArrayList<>();
    }

    public List<Item> getItemsForCategory(String categoryLabel){
        return new ArrayList<>();
    }
}

File: app/src/main/java/com/handstandsam/shoppingapp/mockaccount/MockAccount.kt
package com.handstandsam.shoppingapp.mockaccount


import com.handstandsam.shoppingapp.models.Category
import com.handstandsam.shoppingapp.models.Item
import com.handstandsam.shoppingapp.models.User

abstract class MockAccount {

    var itemsByCategory: MutableMap<String, MutableList<Item>?> = mutableMapOf()

    abstract fun getUsername(): String

    abstract fun getUser(): User

    abstract fun getCategories(): List<Category>

    open fun getItemsForCategory(categoryLabel: String): MutableList<Item>? {
        return mutableListOf()
    }
}
